app/presenters/admin_categories_presenter.rb

```diff
@@ -0,0 +1,69 @@
+class AdminCategoriesPresenter
+  ARTICLES_PER_PAGE = 10
+  attr_reader :category_translation, :per_page
+
+  def initialize(language_key, category_key, per_page = ARTICLES_PER_PAGE)
+    @language_key = language_key
+    @category_translation = CategoryTranslation.translation_for(category_key, @language_key)
+    @per_page = per_page
+
+    unless @category_translation
+      raise ArgumentError.new "No category translation found by #{category_key} and #{@language_key}"
+    end
+  end
+
+  def articles
+    @articles ||= ArticleTranslation.left_joins(:language, :article)
+                                    .where(*article_translation_query)
+                                    .left_joins(article_join_params)
+                                    .select(selected_fields)
+                                    .order(created_at: :desc)
+
+  end
+
+  def filtered_articles(params)
+    final_filter_query = {}
+    final_filter_query[:conferences] = {id: params[:conference]} if params[:conference].present?
+    final_filter_query[:teams] = {id: params[:team]} if params[:team].present?
+    final_filter_query[:status] = params[:published] if params[:published].present?
+
+    page = params[:page] ? params[:page].to_i : 0
+
+    final_filter_query.empty? ? with_offset(articles, page) : with_offset(articles.where(final_filter_query), page)
+  end
+
+  def conferences_hash
+    articles.map{|article| [article.conference_name, article.conference_id]}.uniq.to_h
+  end
+
+  def teams_hash(conference_id = nil)
+    if conference_id && conference_id.to_s.strip != ''
+      articles_array = articles.to_a.select { |article| article.conference_id == conference_id.to_i }
+    else
+      articles_array = articles.to_a
+    end
+    articles_array.map{|article| [article.team_name, article.team_id]}.uniq.to_h
+  end
```

Comment:
```ruby
def teams_hash(conference_id = nil)
  if conference_id.present?
    collection = articles.where(conference_id: conference_id)
  else
    collection = articles
  end

  collection.pluck(:team_name, :team_id).uniq.to_h
end
```

<img width="385" alt="Screenshot 2021-11-30 at 13 05 13" src="https://user-images.githubusercontent.com/16745269/144036516-5ede97ed-8e7a-473a-a22f-7ff58477fbfe.png">

Reply:
It is incorrect to use where here as it will fire the additional query to the DB while the select `.to_a.select` will not